reconcile/utils/ocm_base_client.py

```diff
@@ -108,3 +146,29 @@ def patch(
     def delete(self, api_path: str):
         r = self._session.delete(f"{self._url}{api_path}", timeout=REQUEST_TIMEOUT_SEC)
         r.raise_for_status()
+
+
+class OCMAPIClientConfiguration(Protocol):
+    url: str
+    access_token_client_id: str
+    access_token_url: str
+    access_token_client_secret: VaultSecret
```

Comment:
Since this is a protocol it might be better to use `reconcile.utils.secret_reader.HasSecret` here? In the end the only requirement here is `secret_reader` compatibility if I understand that right.

Reply:
good point. thank you. fixed!